addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiBridgeHandler.java

```diff
@@ -0,0 +1,91 @@
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.nuki.NukiBindingConstants;
+import org.openhab.binding.nuki.dataexchange.BridgeInfoResponse;
+import org.openhab.binding.nuki.dataexchange.NukiHttpClient;
+import org.openhab.binding.nuki.dataexchange.NukiHttpServer;
+import org.openhab.binding.nuki.dataexchange.NukiHttpServerListener;
+import org.openhab.binding.nuki.dto.BridgeApiLockStateRequestDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NukiBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiBridgeHandler extends BaseBridgeHandler implements NukiHttpServerListener {
+
+    private final static Logger logger = LoggerFactory.getLogger(NukiBridgeHandler.class);
+
+    private NukiHttpServer nukiHttpServer;
+
+    public NukiBridgeHandler(Bridge bridge) {
+        super(bridge);
+        logger.trace("Instantiating NukiBridgeHandler({})", bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("NukiBridgeHandler:initialize()");
+        nukiHttpServer = NukiHttpServer.getInstance(this.getConfig(), this);
+        BridgeInfoResponse bridgeInfoResponse = new NukiHttpClient(this.getConfig()).getBridgeInfo();
+        if (bridgeInfoResponse.getStatus() == 200) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                    "Found " + bridgeInfoResponse.getBridgeInfo().getScanResults().size() + " Nuki Smart Locks.");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, bridgeInfoResponse.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiBridgeHandler:handleCommand({}, {})", channelUID, command);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiBridgeHandler:dispose");
+        nukiHttpServer.stopServer();
+    }
+
+    @Override
+    public void handleBridgeLockStateChange(BridgeApiLockStateRequestDto bridgeApiLockStateRequestDto) {
+        logger.debug("NukiBridgeHandler:handleBridgeLockStateChange({})", bridgeApiLockStateRequestDto);
+        String nukiId = String.valueOf(bridgeApiLockStateRequestDto.getNukiId());
+        Bridge bridge = this.getThing();
+        List<Thing> things = bridge.getThings();
+        for (Thing thing : things) {
+            String nukiIdThing = (String) thing.getConfiguration().get(NukiBindingConstants.CONFIG_NUKIID);
+            if (nukiId.equals(nukiIdThing)) {
+                Channel channel = thing.getChannel(NukiBindingConstants.CHANNEL_SMARTLOCKOPENCLOSE);
+                State state = bridgeApiLockStateRequestDto.getState() == 1 ? OnOffType.ON : OnOffType.OFF;
```

Comment:
I'll stick with OnOffType because I use the Smart Lock with a Switch item and therefore an OnOffType is needed.
An OpenClosedType is not allowed for a Switch item.
I also tried to define a Contact as item and a Switch in the sitemap which did not work either.